src/amuse/ext/orbital_elements.py

```diff
@@ -27,81 +28,228 @@ def newton(f,x0,fprime=None,args=(),tol=1.48e-8,maxiter=50):
 
 
 def true_anomaly_from_eccentric_anomaly(E,e):
-  return 2*numpy.arctan2((1+e)**0.5*numpy.sin(E/2),(1-e)**0.5*numpy.cos(E/2))
+    return 2*arctan2((1+e)**0.5*sin(E/2),(1-e)**0.5*cos(E/2))
 
 # E from M,e
 # newton solver for M=E-e sin E
 
-def new_binary_from_orbital_elements(
-        mass1,
-        mass2,
-        semimajor_axis, 
-        eccentricity = 0,
-        true_anomaly = 0, 
-        inclination = 0,
-        longitude_of_the_ascending_node = 0,
-        argument_of_periapsis = 0,
-        G=nbody_system.G
-    ):
-    """ 
-
-    Function that returns two-particle Particle set, with the second 
-    particle position and velocities computed from the input orbital 
-    elements. angles in degrees, inclination between 0 and 180
-
-    """
-    
-    inclination = numpy.radians(inclination)
-    argument_of_periapsis = numpy.radians(argument_of_periapsis)
-    longitude_of_the_ascending_node = numpy.radians(longitude_of_the_ascending_node)
-    true_anomaly = numpy.radians(true_anomaly)
-
-    cos_true_anomaly = numpy.cos(true_anomaly)
-    sin_true_anomaly = numpy.sin(true_anomaly)    
-
-    cos_inclination = numpy.cos(inclination)
-    sin_inclination = numpy.sin(inclination)    
 
-    cos_arg_per = numpy.cos(argument_of_periapsis)
-    sin_arg_per = numpy.sin(argument_of_periapsis)
+def angle_with_unit(angle, default_unit=units.deg):
+    try:
+        unit = angle.unit
+    except:
+        angle = angle | default_unit
+    return angle
 
-    cos_long_asc_nodes = numpy.cos(longitude_of_the_ascending_node)
-    sin_long_asc_nodes = numpy.sin(longitude_of_the_ascending_node)
 
-    ### alpha is a unit vector directed along the line of node ###
-    alphax = cos_long_asc_nodes*cos_arg_per - sin_long_asc_nodes*sin_arg_per*cos_inclination
-    alphay = sin_long_asc_nodes*cos_arg_per + cos_long_asc_nodes*sin_arg_per*cos_inclination
+def equal_length_array_or_scalar(
```

Comment:
not sure i like this one - or at least its default behaviour (ie taking the first value if the lengths are unequal). if the length are unequal an exception should be raised (but then that is the default behaviour...)...but maybe I am not seeing a particular case where this is handy...